test/functional/http/post/4.multisig.js

```diff
@@ -0,0 +1,530 @@
+'use strict';
+
+var node = require('../../../node');
+var shared = require('../../shared');
+var constants = require('../../../../helpers/constants');
+
+var sendTransactionPromise = require('../../../common/apiHelpers').sendTransactionPromise;
+var creditAccountPromise = require('../../../common/apiHelpers').creditAccountPromise;
+var sendSignaturePromise = require('../../../common/apiHelpers').sendSignaturePromise;
+var waitForConfirmations = require('../../../common/apiHelpers').waitForConfirmations;
+
+describe('POST /api/transactions (type 4) register multisignature', function () {
+
+	var scenarios = {
+		'no_funds': new shared.multisigScenario(3, 0),
+		'minimal_funds': new shared.multisigScenario(3, constants.fees.multisignature * 3),
+		'minimum_not_reached': new shared.multisigScenario(4), //4_2
+		'regular': new shared.multisigScenario(3), //3_2
+		'max_signatures': new shared.multisigScenario(constants.multisigConstraints.keysgroup.maxItems + 1), //16_2
+		'max_signatures_max_min': new shared.multisigScenario(constants.multisigConstraints.keysgroup.maxItems + 1), //16_16
+		'more_than_max_signatures': new shared.multisigScenario(constants.multisigConstraints.keysgroup.maxItems + 2), //17_2
+	};
+
+	var transaction, signature;
+	var transactionsToWaitFor = [];
+	var badTransactions = [];
+	var goodTransactions = [];
+	var badTransactionsEnforcement = [];
+	var goodTransactionsEnforcement = [];
+	var pendingMultisignatures = [];
+
+	before(function () {
+		//Crediting accounts
+		return node.Promise.all(Object.keys(scenarios).map(function (key) {
+
+			if (key === 'no_funds') { return; }
+
+			return creditAccountPromise(scenarios[key].account.address, scenarios[key].amount).then(function (res) {
+				node.expect(res).to.have.property('success').to.be.ok;
+				node.expect(res).to.have.property('transactionId').that.is.not.empty;
+				transactionsToWaitFor.push(res.transactionId);
+			});
+		}))
+			.then(function (res) {
+				return waitForConfirmations(transactionsToWaitFor);
+			});
+	});
+
+	describe('schema validations', function () {
+
+		shared.invalidAssets(scenarios.regular.account, 'multisignature', badTransactions);
+
+		describe('keysgroup', function () {
+
+			it('using empty array should fail', function () {
```

Comment:
Consider adding extra `describe` blocks 
```
describe('keysgroup')
   describe('regular account')
   ...
   describe('17_2 account')
   ...
```

Reply:
Here and in the following blocks - min, lifetime, transaction processing, signing transaction ...

Reply:
By now it is not feasible. We should rethink the whole approach later on when having the final picture after porting 0.9.10.